### PR TITLE
add subset, offset and size index

### PR DESCRIPTION
**Pull request by anjor**

### CODE REVIEW

File: indexes/subset-offset-and-size.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package indexes
+
+import "errors"
+
+type SubsetOffsetAndSize struct {
+	Subset uint64 // uint24, 3 bytes, max 16.7 MB (megabytes)
```
Comment (linuskendall) [inline indexes/subset-offset-and-size.go:6]: could be even smaller? uint16 ? but might not be worth it and better to have an extra byte

**Review by linuskendall — APPROVED** [review]

File: indexes/subset-offset-and-size.go (hunk 1 above)
Comment (anjor) [reply indexes/subset-offset-and-size.go:6]: Yeah I don't have a strong opinion here. @gagliardetto ?

Comment (gagliardetto) [reply indexes/subset-offset-and-size.go:6]: yeah uint16 should be eough

File: cmd-x-index-cid2subsetoffset.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	"k8s.io/klog/v2"
+)
+
+func newCmd_Index_cid2subsetOffset() *cli.Command {
+	var verify bool
+	var epoch uint64
+	var network indexes.Network
+	var indexDir string
+	return &cli.Command{
+		Name:        "cid-to-offset",
```
Comment (gagliardetto) [inline cmd-x-index-cid2subsetoffset.go:20]: this probably needs a different name

**Comment by gagliardetto** [comment]
given that we have the metadata that states the byte beginning and end of split piece, we can avoid creating this new index and just use the existing `cid-to-offset-and-size`

there will be an extra step that will calculate the subset file from an offset.

**Comment by gagliardetto** [comment]
@linuskendall @anjor

**Comment by anjor** [comment]
> given that we have the metadata that states the byte beginning and end of split piece

Do we have this @gagliardetto ?

**Comment by anjor** [comment]
Ohhh you are saying we should add it -- https://github.com/rpcpool/yellowstone-faithful/pull/132#discussion_r1745448674

**Comment by anjor** [comment]
I suspect we are not going to need this after merging https://github.com/rpcpool/yellowstone-faithful/pull/132 

Basically the existing indexes will "just work"

**Pull request closed without merging**